bang/magicollections.py

```diff
@@ -1897,6 +1897,24 @@ def to_fields(self, view, item, *args, **kwargs):
         setSubField(fields, 'length', key='value', value=lambda f: item.length_in_minutes)
         setSubField(fields, 'unlock', key='value', value=item.unlock_sentence)
 
+        for difficulty, verbose_name in models.Song.DIFFICULTIES:
+            note_image = u'{}img/note.png'.format(RAW_CONTEXT['static_url'])
+            image = lambda f: u'{static_url}img/songs/{difficulty}.png'.format(
+                static_url=RAW_CONTEXT['static_url'],
+                difficulty=difficulty,
+            )
+            diff = getattr(item, u'{}_difficulty'.format(difficulty), None)            
+            setSubField(fields, u'{}_notes'.format(difficulty), key='image', value=image)            
+            if diff != None:
+                setSubField(fields, u'{}_difficulty'.format(difficulty), key='image', value=image)
+                setSubField(fields, u'{}_difficulty'.format(difficulty), key='type',
+                            value='html')
+                setSubField(fields, u'{}_difficulty'.format(difficulty), key='value',
+                    value=mark_safe(u'{big_images}{small_images}'.format(diff=diff,
+                    big_images=(u'<img src="{}" class="song-big-note">'.format(note_image) * (diff // 5)),
+                    small_images=(u'<img src="{}" class="song-small-note">'.format(note_image) * (diff % 5)),
```

Comment:
Instead of copying the logic to create the HTML string to show the difficulty, you can create a separate function (in `utils.py` or inside the SongCollection object starting with a `_` to show it's internal). This way, if we decide to change how it's displayed, we only need to change it once.

Reply:
✅